plugins/docker_binary_builder/app/models/binary_builder.rb

```diff
@@ -0,0 +1,136 @@
+class BinaryBuilder
+  DOCKER_BUILD_FILE = 'Dockerfile.build'.freeze
+  BUILD_SCRIPT = '/app/build.sh'.freeze
+  ARTIFACTS_FILE = 'artifacts.tar'.freeze
+  ARTIFACTS_FILE_PATH = "/app/#{ARTIFACTS_FILE}".freeze
+  DOCKER_HOST_CACHE_DIR = '/opt/samson_build_cache'.freeze
```

Comment:
How much disk space are we talking about per project?

I'm not so concerned about a developer with bad intentions. More like a developer who writes build scripts that work fine on their local instance, but starts breaking when running on Samson, with a dozen other projects using the same cache directory.  Or their build scripts start causing some other project to accidentally have problems.

This functionality should work with any project that can be compiled, right?  So it might work totally fine for Java-based projects using .m2 and .ivy.  But maybe it will cause problems when compiling code for Go, or C++, or some other language.

Also, what happens when Samson is running two builds at the same time that are both reading and writing files into the same directory?  That seems like another recipe for problems.

The way I see it, disk space is cheap.  I don't want people on the Ops or Infra teams spending a bunch of time hunting down deploy problems caused by this.

That being said, it would be a pretty simple change if we did find this was causing problems.  So I'm willing to give it a try.  But the first time we have a build problem that is caused by the shared directory, I'm going to refactor it.  :-)